pkg/asset/installconfig/gcp/validation.go

```diff
@@ -32,6 +33,22 @@ func validateProject(client API, ic *types.InstallConfig, fieldPath *field.Path)
 		if _, found := projects[ic.GCP.ProjectID]; !found {
 			return append(allErrs, field.Invalid(fieldPath.Child("project"), ic.GCP.ProjectID, "invalid project ID"))
 		}
+
+	}
+	return allErrs
+}
+
+func validatePublicZoneID(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.PublicZoneID != "" {
+		zone, err := client.GetPublicDNSZoneByID(context.TODO(), ic.GCP.ProjectID, ic.GCP.PublicZoneID)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("publicZoneID"), ic.GCP.PublicZoneID, err.Error()))
+		}
+		if zone.Name != ic.GCP.PublicZoneID || zone.DnsName != ic.BaseDomain {
+			return append(allErrs, field.Invalid(fieldPath.Child("publicZoneID"), ic.GCP.PublicZoneID, fmt.Sprintf("invalid zone id: %s", ic.GCP.PublicZoneID)))
+		}
```

Comment:
1. the error message doesn't need to include the value, it is already part of the field.Invalid(path, value, error/message)
2. the message should be more useful to the user,  like DNS Name for  DNS zone doesn not match basedomain "<basedomain>"

Reply:
not important but would be nice.

Reply:
Fixed!